muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/tabs.tsx

```diff
@@ -38,7 +39,39 @@ interface MobileOnlyTabsProps {
 
 export const Tabs: React.FC<TabsProps> = (props) => {
 
+  const [currentWidth, setCurrentWidth] = React.useState(Math.round(window.innerWidth / 16));
   const { modifier, renderAllComponents, activeTab, onTabChange, tabs, children, allTabs } = props;
+  const mobileBreakpoint = parseInt(variables.mobileBreakpoint); //Parse a breakpoint from scss to a number
+
+  const countRef = React.useRef(currentWidth); // We need this, otherwise the useEffect won't get the correct currentWidth
+  countRef.current = currentWidth;
+
+  React.useEffect(() => {
+    /**
+     * A Handler for the resize event
+     *
+     * @returns
+     */
+    const handleResize = () => () => {
+      const width = Math.round(window.innerWidth / 16); // Width on resize
+      const direction = countRef.current < width ? "out" : "in"; // Direction of the resize
+
+      /** If the resize zoom direction is outwards
+       * we don't need to re-render
+       * if we're already using the correct component
+       * and vice versa
+       */
+      if ((direction === "out" && countRef.current <= mobileBreakpoint)
+        || (direction === "in" && countRef.current >= mobileBreakpoint)) {
+        setCurrentWidth(width);
+      }
+    }
+
+    window.addEventListener("resize", handleResize());
+    return () => window.removeEventListener("resize", handleResize());
+  }, []);
```

Comment:
This whole useEffect block could be remade as custom hook by naming it useIsMobileWidth for example.